articles/consuming-apis-in-flutter/index.md

```diff
@@ -0,0 +1,250 @@
+
+```dart
+import 'package:HTTP/HTTP.dart' as HTTP;
+```
+
+The below shows how we will use the package to fetch data from the API.
+
+```dart
+//importing HTTP package for fetching and consuming HTTP resources
+import 'package:HTTP/HTTP.dart' as HTTP;
+
+//Github request class
+class Github {
+  final String userName;
+  final String url = 'https://api.github.com/';
+  static String clientId = 'CLIENT_ID';
+  static String clientSecret = 'CLIENT_SECRET';
+
+  //Github class constructor
+  Github(this.userName);
+
+  //Fetch a user with the username supplied in the form input
+  Future<http.Response> fetchUser() {
+    return http.get(url + 'users/' + userName);
+  }
+
+  //Fetch repositories of the given user
+  Future<http.Response> fetchRepos() {
+    return http.get(url + 'users/' + userName + '/repos');
+  }
+}
+```
+
+### Creating Data classes from JSON
+Since flutter accepts dart as the primary programming language, we need to convert the JSON data fetched from the URL to dart Classes for consumption in the application. We can do that using the [Quicktype](https://app.quicktype.io/) website where we pass the JSON object, and a class of the object is returned based on a specified language. Will will return our classes in `dart`. For instance, our JSON representing the User is as shown below:
+
+```dart
+{
+  "login": "jerimkaura",
+  "avatar_url": "https://avatars.githubusercontent.com/u/50904889?v=4",
+  "location": "Nairobi"
+}
+```
+
+#### The user class
+I edited the JSON to capture only the attributes needed on the application. When we pass the above JSON into [Quicktype](https://app.quicktype.io/), the generated user class is as below:
+
+```dart
+// To parse this JSON data, do
+//     final user = userFromJson(jsonString);
+
+import 'dart:convert';
+
+//the created user class
+class User {
+  //class constructor
+  User({
+    this.login,
+    this.avatarUrl,
+    this.location,
+  });
+
+  String login;
+  String avatarUrl;
+  String location;
+
+  //JSON serialization
+  factory User.fromRawJson(String str) => User.fromJson(json.decode(str));
+
+  String toRawJson() => json.encode(toJson());
+
+  factory User.fromJson(Map<String, dynamic> json) => User(
+    login: json["login"],
+    avatarUrl: json["avatar_url"],
+    location: json["location"],
+  );
+
+  Map<String, dynamic> toJson() => {
+    "login": login,
+    "avatar_url": avatarUrl,
+    "location": location,
+  };
+}
+```
+
+### Adding the providers
+The Provider will have the functions required to fetch the API's user data and deliver a response. We will create a file called `UserProvider.dart` under the `Providers` folder. The `ChangeNotifier` class will notify our view when one more variable changes.
+
+```dart
+
+class UserProvider with ChangeNotifier {
+  User user; //an instance of a user
+  String errorMessage; //error message
+  bool loading = false; //loading the page
+
+  Future<bool> fetchUser(username) async {
```

Comment:
it might be a good idea to explain why we are using an Async method here.
Also, please include more code comments that effectively explains all the lines